crates/runtime/src/dataconnector/github/mod.rs

```diff
@@ -397,53 +402,133 @@ pub fn parse_globs(input: &str) -> super::DataConnectorResult<Arc<GlobSet>> {
+            "state",
+            GitHubPushdownSupport {
+                ops: vec![Operator::Eq, Operator::NotEq],
+                remap: None,
+            },
+        );
+
+        m.insert(
+            "body",
+            GitHubPushdownSupport {
+                ops: vec![
+                    Operator::Eq,
+                    Operator::NotEq,
+                    Operator::LikeMatch,
+                    Operator::ILikeMatch,
+                    Operator::NotLikeMatch,
+                    Operator::NotILikeMatch,
+                ],
+                remap: None,
+            },
+        );
+
+        m
+    };
+}
 
 pub(crate) fn filter_pushdown(expr: &Expr) -> FilterPushdownResult {
-    if let Expr::BinaryExpr(binary_expr) = expr {
-        let ((Expr::Literal(value), Expr::Column(column))
-        | (Expr::Column(column), Expr::Literal(value))) =
-            (*binary_expr.left.clone(), *binary_expr.right.clone())
-        else {
-            return FilterPushdownResult {
-                filter_pushdown: TableProviderFilterPushDown::Unsupported,
-                expr: expr.clone(),
-                context: None,
+    let column_matches = match expr {
+        Expr::BinaryExpr(binary_expr) => {
+            match (*binary_expr.left.clone(), *binary_expr.right.clone()) {
+                (Expr::Column(column), Expr::Literal(value))
+                | (Expr::Literal(value), Expr::Column(column)) => {
+                    Some((column, value, binary_expr.op))
+                }
+                _ => None,
+            }
+        }
+        Expr::Like(like_expr) => match (*like_expr.expr.clone(), *like_expr.pattern.clone()) {
+            (Expr::Column(column), Expr::Literal(value))
+            | (Expr::Literal(value), Expr::Column(column)) => {
+                let op = match (like_expr.negated, like_expr.case_insensitive) {
+                    (false, false) => Operator::LikeMatch,
+                    (true, false) => Operator::NotLikeMatch,
+                    (false, true) => Operator::ILikeMatch,
+                    (true, true) => Operator::NotILikeMatch,
+                };
+
+                Some((column, value, op))
+            }
+            _ => None,
+        },
+        _ => None,
+    };
+
+    if let Some((column, value, op)) = column_matches {
+        if let Some(column_support) = GITHUB_FILTER_PUSHDOWNS_SUPPORTED.get(column.name.as_str()) {
+            if !column_support.ops.contains(&op) {
+                tracing::debug!("Unsupported operator {op} for column {}", column.name);
+
+                return FilterPushdownResult {
+                    filter_pushdown: TableProviderFilterPushDown::Unsupported,
+                    expr: expr.clone(),
+                    context: None,
+                };
+            }
+
+            let value = if let ScalarValue::Utf8(Some(v)) = &value {
+                match column.name.as_str() {
+                    "state" => ScalarValue::Utf8(Some(v.to_lowercase())),
```

Comment:
@peasee  - this seem just `"state" => value`.
Why 'state' is special and we process it specially?

Reply:
yes, GitHub search only searches using the lowercase value. They also combine `merged` and `closed` pull requests under the same search term for `closed`.